src/rgw/rgw_user.cc

```diff
@@ -1391,13 +1391,18 @@ int RGWSubUserPool::add(RGWUserAdminOpState& op_state, std::string *err_msg, boo
 {
   std::string subprocess_msg;
   int ret;
+  int32_t key_type = op_state.get_key_type();
 
   ret = check_op(op_state, &subprocess_msg);
   if (ret < 0) {
     set_err_msg(err_msg, "unable to parse request, " + subprocess_msg);
     return ret;
   }
 
+  if (key_type == KEY_TYPE_S3 && op_state.get_access_key().empty()) {
```

Comment:
Yeah,but the  method you said "RGWAccessKeyPool::generate_key(...)" just add access key for SWIFT type user or S3 type user that the gen_access = true .The original codes has not set gen_access = true when access_key.empty() =true as creating  a subuser .So it is neccessary to check KEY_TYPE here .